corehq/apps/user_importer/helpers.py

```diff
@@ -9,3 +12,60 @@ def spec_value_to_boolean_or_none(user_spec_dict, key):
         return value
     else:
         return None
+
+
+class UserChangeLogger(object):
+    def __init__(self, domain, user, is_new_user, changed_by_user, changed_via):
+        self.domain = domain
+        self.user = user
+        self.is_new_user = is_new_user
+        self.changed_by_user = changed_by_user
+        self.changed_via = changed_via
+
+        self.original_user_doc = self.user.to_json()
+        self.fields_changed = {}
+        self.messages = []
+
+    def add_changes(self, changes):
+        # ignore for new user since the whole user doc is logged for a new user
```

Comment:
Yes, you are right. It should not be needed.

Would like to hear thoughts on this.
Since the copy of the doc is kept in `init`, and it isn't a costly operation, how about keeping it simple, without the `if` block and then leave the methods to decide if they need it, leaving the awareness of distinction in operations for new/existing objects, only where needed.

Reply:
It may not be that expensive as a single operation but since this is part of an import it needs to be done for each user being updated and the cost could add up. I'd be pro not setting this prop (or setting it to None) for new users.

Reply:
https://github.com/dimagi/commcare-hq/pull/29935/commits/07746469f1d10d6907dee5ca1d547c6f3eb71cb0